jscomp/core/js_of_lam_variant.ml

```diff
@@ -84,3 +84,13 @@ let eval_as_int (arg : J.expression) (dispatches : (int * int) list ) : E.t  =
                body = [S.return (E.int (Int32.of_int  r))],
                       false (* FIXME: if true, still print break*)
               }) dispatches))]
+
+let eval_as_unwrap (arg : J.expression) : E.t =
+  if arg == E.undefined then
+    E.undefined
+  else
```

Comment:
can you add a ffi failed test like this
https://github.com/BuckleScript/bucklescript/blob/master/jscomp/ounit_tests/ounit_ffi_error_debug_test.ml#L33
run `make snapshotml && make test`